internal/project/project.go

```diff
@@ -81,981 +49,315 @@ const (
-	// } else {
-	// 	this.resolutionCache.invalidateResolutionOfFile(info.path);
-	// }
-	// this.cachedUnresolvedImportsPerFile.delete(info.path);
-	p.markAsDirtyLocked()
-}
-
-func (p *Project) AddInferredProjectRoot(info *ScriptInfo) {
-	p.mu.Lock()
-	defer p.mu.Unlock()
-	if p.isRoot(info) {
-		panic("script info is already a root")
-	}
-	p.setRootFileNameOfInferred(info.path, info.fileName)
-	p.programConfig = nil
-	p.setTypeAcquisition(nil)
-	// !!!
-	// if p.kind == KindInferred {
-	// 	p.host.startWatchingConfigFilesForInferredProjectRoot(info.path);
-	// }
-	info.attachToProject(p)
-	p.markAsDirtyLocked()
-}
-
-func (p *Project) LoadConfig() error {
-	if p.kind != KindConfigured {
-		panic("loadConfig called on non-configured project")
-	}
-
-	p.programConfig = nil
-	p.pendingReload = PendingReloadNone
-	p.parsedCommandLine = p.GetResolvedProjectReference(p.configFileName, p.configFilePath)
-	if p.parsedCommandLine != nil {
-		p.Logf("Config: %s : %s",
-			p.configFileName,
-			core.Must(core.StringifyJson(map[string]any{
-				"rootNames":         p.parsedCommandLine.FileNames(),
-				"options":           p.parsedCommandLine.CompilerOptions(),
-				"projectReferences": p.parsedCommandLine.ProjectReferences(),
-			}, "    ", "  ")),
+		newProgram = compiler.NewProgram(
+			compiler.ProgramOptions{
+				Host:                        p.host,
+				Config:                      commandLine,
+				UseSourceOfProjectReference: true,
+				TypingsLocation:             typingsLocation,
+				JSDocParsingMode:            ast.JSDocParsingModeParseAll,
+				CreateCheckerPool: func(program *compiler.Program) compiler.CheckerPool {
+					checkerPool = newCheckerPool(4, program, p.log)
+					return checkerPool
+				},
+			},
 		)
-
-		p.compilerOptions = p.parsedCommandLine.CompilerOptions()
-		p.setTypeAcquisition(p.parsedCommandLine.TypeAcquisition())
-		p.setRootFiles(p.parsedCommandLine.FileNames())
-	} else {
-		p.compilerOptions = &core.CompilerOptions{}
-		p.setTypeAcquisition(nil)
-		return fmt.Errorf("could not read file %q", p.configFileName)
-	}
-	return nil
-}
-
-// setRootFiles returns true if the set of root files has changed.
-func (p *Project) setRootFiles(rootFileNames []string) {
-	newRootScriptInfos := make(map[tspath.Path]struct{}, len(rootFileNames))
-	for _, file := range rootFileNames {
-		path := p.toPath(file)
-		// !!! updateNonInferredProjectFiles uses a fileExists check, which I guess
-		// could be needed if a watcher fails?
-		newRootScriptInfos[path] = struct{}{}
-		p.rootFileNames.Set(path, file)
-		// if !isAlreadyRoot {
-		// 	if scriptInfo.isOpen {
-		// 		!!!s.removeRootOfInferredProjectIfNowPartOfOtherProject(scriptInfo)
-		// 	}
-		// }
-	}
-
-	if p.rootFileNames.Size() > len(rootFileNames) {
-		for root := range p.rootFileNames.Keys() {
-			if _, ok := newRootScriptInfos[root]; !ok {
-				p.rootFileNames.Delete(root)
-			}
+		if p.Program != nil && p.Program.HasSameFileNames(newProgram) {
```

Comment:
nit change to program.HasSameFileNames to HasSameFiles ? as its checking that the files themselves are same too?